python/lsst/daf/butler/registry/wildcards.py

```diff
@@ -839,6 +839,12 @@ def iter(
                                         flattenChains=flattenChains, includeChains=includeChains):
             yield record
 
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, CollectionQuery):
+            return self._search == other._search and self._patterns == other._patterns
+        else:
```

Comment:
The else doesn't really do much so could be dropped and just have the return as the last line in the method.

Reply:
I've seen style guides (in other languages, I think?) that recommend this form to "preserve the symmetry of the branching" or some such, and I have never had a good sense for whether one form other the other was idiomatic in Python.  But if this was strange enough to you to merit a comment, it probably merits changing.